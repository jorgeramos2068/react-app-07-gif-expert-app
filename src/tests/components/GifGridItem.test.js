import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Tests for <GifGridItem />', () => {
  let wrapper;
  const title = 'Test title';
  const url = 'http://localhost/test.png'

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test('Should show <GifGridItem /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should display a <p> with title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Should display an image with url and alt', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Should contain animate classes', () => {
    const containerDiv = wrapper.find('div');
    expect(containerDiv.prop('className').includes('animate__fadeIn')).toBe(true);
  });
});
