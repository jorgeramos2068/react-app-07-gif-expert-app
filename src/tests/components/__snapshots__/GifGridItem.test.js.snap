// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem /> Should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Test title"
    src="http://localhost/test.png"
  />
  <p>
    Test title
  </p>
</div>
`;
